jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -197,6 +197,12 @@ static long getValueFieldOffset() {
 			}
 		}
 	}
+	
+	/**
+	 * This is a System property to enable copying of the underlying value array in {@link #String.substring(int)} and 
+	 * {@link #String.substring(int, int)} if the offset is zero.
+	 */
+	static boolean enableCopyInSubstringWhenOffsetIsZero;
```

Comment:
We get more streamlined code if we capture `enableSharingInSubstringWhenOffsetIsZero' in that it doesn't need to be negated in every use case.

Reply:
Fixed in a44d52e.